packages/transport/src/modules/sync/externalEventProcessors/ExternalEventProcessorRegistry.ts

```diff
@@ -0,0 +1,45 @@
+import { EventBus } from "@js-soft/ts-utils";
+import { TransportError } from "../../../core";
+import { MessageController } from "../../messages/MessageController";
+import { RelationshipsController } from "../../relationships/RelationshipsController";
+import { ChangedItems } from "../ChangedItems";
+import { ExternalEventProcessor } from "./ExternalEventProcessor";
+import { MessageDeliveredExternalEventProcessor } from "./MessageDeliveredExternalEventProcessor";
+import { MessageReceivedExternalEventProcessor } from "./MessageReceivedExternalEventProcessor";
+import { RelationshipChangeCompletedExternalEventProcessor } from "./RelationshipChangeCompletedExternalEventProcessor";
+import { RelationshipChangeCreatedExternalEventProcessor } from "./RelationshipChangeCreatedExternalEventProcessor";
+
+export class ExternalEventProcessorRegistry {
+    private readonly processors = new Map<string, ExternalEventProcessor>();
+    public constructor(
+        eventBus: EventBus,
+        changedItems: ChangedItems,
+        ownAddress: string,
+        messagesController: MessageController,
+        relationshipsController: RelationshipsController
+    ) {
+        this.registerProcessor("MessageReceived", new MessageReceivedExternalEventProcessor(eventBus, changedItems, ownAddress, messagesController));
+        this.registerProcessor("MessageDelivered", new MessageDeliveredExternalEventProcessor(eventBus, changedItems, ownAddress, messagesController));
+        this.registerProcessor("RelationshipChangeCreated", new RelationshipChangeCreatedExternalEventProcessor(eventBus, changedItems, ownAddress, relationshipsController));
+        this.registerProcessor("RelationshipChangeCompleted", new RelationshipChangeCompletedExternalEventProcessor(eventBus, changedItems, ownAddress, relationshipsController));
```

Comment:
Yea, but as the constructors are not the same for some of the Processors. This will get a bit tricky. We either add all the controllers to all Processors also for future Events or leave it as it currently is.

Reply:
You can align the ctors

Reply:
See my other comment: https://github.com/nmshd/runtime/pull/86#discussion_r1551708411
If we pass the AccountController, we don't need any other controllers anymore.